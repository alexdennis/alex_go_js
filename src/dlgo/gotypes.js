export const PLAYER_BLACK = 1;
export const PLAYER_WHITE = 2;

export function other(player) {
  return player === PLAYER_BLACK ? PLAYER_WHITE : PLAYER_BLACK;
}

export class Point {
  constructor(row, col) {
    this.row = row;
    this.col = col;
  }

  neighbors = () => [
    new Point(this.row - 1, this.col),
    new Point(this.row + 1, this.col),
    new Point(this.row, this.col - 1),
    new Point(this.row, this.col + 1)
  ];
}
