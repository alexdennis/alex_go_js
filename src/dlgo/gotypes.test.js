import { other, PLAYER_BLACK, PLAYER_WHITE, Point } from "./gotypes";

test("other give me the other player", () => {
  expect(other(PLAYER_BLACK)).toBe(PLAYER_WHITE);
  expect(other(PLAYER_WHITE)).toBe(PLAYER_BLACK);
});

describe("Point", () => {
  test("constructs", () => {
    const point = new Point(0, 0);
    expect(point.col).toBe(0);
    expect(point.row).toBe(0);
  });

  test("can tell its neighbors", () => {
    const point = new Point(3, 4);
    expect(point.neighbors()).toMatchSnapshot();
  });
});
