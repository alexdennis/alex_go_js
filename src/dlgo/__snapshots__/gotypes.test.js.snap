// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Point can tell its neighbors 1`] = `
Array [
  Point {
    "col": 4,
    "neighbors": [Function],
    "row": 2,
  },
  Point {
    "col": 4,
    "neighbors": [Function],
    "row": 4,
  },
  Point {
    "col": 3,
    "neighbors": [Function],
    "row": 3,
  },
  Point {
    "col": 5,
    "neighbors": [Function],
    "row": 3,
  },
]
`;
